cli/golem-templates/templates/rust/rust-app-common/.vscode/settings.json

```diff
@@ -1,3 +1,3 @@
 {
-    "rust-analyzer.server.extraEnv": { "CARGO": "cargo-component" }
+    "rust-analyzer.procMacro.enable": true
```

Comment:
This settings with server.extraEnv as cargo component was slowing down builds within vs-code by 3-4 minutes, and sometimes never finishes. Hence I changed it. This is probably happening after we made it code first.

Also tried to re-install vs-code again just to see and  I had instances where I had to manually set this. So I choose to make it part of the template. Safer. We don't want red things in their IDE.

I tested with rust-rover as part of this testing: For rust-rover, new installations of rust-rover has good defaults. Macro expansions are on by default. I think we won't have instances of false compiler errors for rust-rover users.